forms-flow-api/src/formsflow_api/resources/application.py

```diff
@@ -386,3 +388,25 @@ def get():
             )
         except BusinessException as err:
             return err.error, err.status_code
+
+@cors_preflight("DELETE, OPTIONS")
+@API.route("/<int:application_id>/delete", methods=["DELETE", "OPTIONS"])
```

Comment:
I wasn't sure how to name this endpoint.
Delete version of `applications/<applicatoinId>` is already used for Camunda deleting application so I added another one `applications/<applicatoinId>/delete`. Feel free to suggest a better naming :)